src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useIsApprovalNeeded.tsx

```diff
@@ -6,24 +6,54 @@ import { useCallback, useMemo } from 'react'
 import { reactQueries } from 'react-queries'
 import { selectAllowanceCryptoBaseUnit } from 'react-queries/hooks/selectors'
 import type { GetAllowanceErr } from 'react-queries/types'
-import { bn } from 'lib/bignumber/bignumber'
+import { usdtAssetId } from 'components/Modals/FiatRamps/config'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import { selectRelatedAssetIdsInclusive } from 'state/slices/related-assets-selectors'
+import { useAppSelector } from 'state/store'
 
 export const useIsApprovalNeeded = (
   tradeQuoteStep: TradeQuoteStep | undefined,
   sellAssetAccountId: AccountId | undefined,
 ) => {
+  const relatedAssetIdsFilter = useMemo(() => ({ assetId: usdtAssetId }), [])
+  const usdtAssetIds = useAppSelector(state =>
+    selectRelatedAssetIdsInclusive(state, relatedAssetIdsFilter),
+  )
+  const isUsdtApprovalResetEnabled = useFeatureFlag('UsdtApprovalReset')
   const selectIsApprovalNeeded = useCallback(
     (data: Result<string, GetAllowanceErr>) => {
       if (tradeQuoteStep === undefined) return undefined
 
       const allowanceCryptoBaseUnit = selectAllowanceCryptoBaseUnit(data)
-      return allowanceCryptoBaseUnit !== undefined
-        ? bn(allowanceCryptoBaseUnit).lt(
-            tradeQuoteStep.sellAmountIncludingProtocolFeesCryptoBaseUnit,
-          )
-        : false
+      const isApprovalNeeded =
+        allowanceCryptoBaseUnit !== undefined
+          ? bn(allowanceCryptoBaseUnit).lt(
+              tradeQuoteStep.sellAmountIncludingProtocolFeesCryptoBaseUnit,
+            )
+          : false
+
+      const isAllowanceResetNeeded = (() => {
+        switch (true) {
+          case !isApprovalNeeded:
+            return false
+          case !isUsdtApprovalResetEnabled:
+            return false
+          case bnOrZero(allowanceCryptoBaseUnit).isZero():
+            return false
+          case !usdtAssetIds.some(usdtAssetId => usdtAssetId === tradeQuoteStep?.sellAsset.assetId):
+            return false
+          default:
+            return true
+        }
+      })()
```

Comment:
nitpick - big switch to just return true for one condition haha
```suggestion
      const isUsdt = usdtAssetIds.some(assetId => assetId === tradeQuoteStep?.sellAsset.assetId)
      const isAllowanceResetNeeded = isUsdtApprovalResetEnabled && isApprovalNeeded && isUsdt
```